corecito.py

```diff
@@ -50,20 +64,45 @@ async def main():
         logger.info(f'> Exploded {increase_percentage:.2f}%\nConsider updating CoreNumber to {deviated_core_number:.6f}')
 
       elif coreNumberIncreased(account.core_number, deviated_core_number, account.min_core_number_increase_percentage, account.max_core_number_increase_percentage):
-        logger.info(f'Increased {increase_percentage:.2f}% - excess of {excess:.6f} {account.core_number_currency} denominated in {account.base_currency}')
-        tx_result = round(excess * buy_price, account.max_decimals_buy)
-        logger.info(f'\n\n>>> Selling: {tx_result:.6f} {account.base_currency} at {buy_price} to park an excess of {excess:.6f} {account.core_number_currency}\n')
+
+        #Check if 'fiat' is True to adjust messages format and tx_result var and 'excess' has to be divided by the buy_price
+        if fiat:
+          logger.info(f'Increased {increase_percentage:.2f}% - excess of €{excess:.2f} denominated in {account.base_currency}')
+          tx_result = round(excess / buy_price, account.max_decimals_buy)
+          logger.info(f'\n\n>>> Selling: {tx_result:.6f} {account.base_currency} at €{buy_price} to park an excess of €{excess:.2f}\n')
+        else:
+          logger.info(f'Increased {increase_percentage:.2f}% - excess of {excess:.6f} {account.core_number_currency} denominated in {account.base_currency}')
+          tx_result = round(excess * buy_price, account.max_decimals_buy)
+          logger.info(f'\n\n>>> Selling: {tx_result:.6f} {account.base_currency} at {buy_price} to park an excess of {excess:.6f} {account.core_number_currency}\n')
+
         # Sell excess of base currency ie. => in ETH_BTC pair, sell excess BTC => Buy ETH
+        # If fiat, we sell the value previously calculated and stored on tx_result
         if (not config['safe_mode_on']):
-          await account.order_market_buy(tx_result, excess)
+          if fiat:
+            await account.order_market_sell(tx_result)
+          else:
+            await account.order_market_buy(tx_result, excess)
 
       elif coreNumberDecreased(account.core_number, deviated_core_number, account.min_core_number_decrease_percentage, account.max_core_number_decrease_percentage):
-        logger.info(f'Decreased {decrease_percentage:.2f}% - missing {missing:.6f} {account.core_number_currency} denominated in {account.base_currency}')
-        tx_result = missing * sell_price
-        logger.info(f'\n\n>>> Buying: {tx_result:.6f} {account.base_currency} at {buy_price} taking {missing:.6f} {account.core_number_currency} from reserves\n')
+        
+        #Check is fiat is True to adjust messages format and tx_result var and 'missing' has to be divided by the sell_price
+        if fiat:
+          logger.info(f'Decreased {decrease_percentage:.2f}% - missing {missing:.6f} {account.core_number_currency} denominated in {account.base_currency}')
+          tx_result = round(missing / sell_price, account.max_decimals_sell)
```

Comment:
Same question as above inverted, shouldn't we use here the buy_price for fiat?

Reply:
Same as before, as we are using Market Orders, you buy to the next Seller (Ask) in line in the Order Book.